moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -45,6 +45,12 @@
 #include <Eigen/Core>
 #include <moveit/robot_model/robot_model.h>
 
+namespace
```

Comment:
What is this namespace for?

Reply:
an anonymous namespace *and* a static definition are somewhat weird indeed.
But the problem is that this exposes a LOGNAME variable *in a common MoveIt header* which it of course mustn't.

Reply:
The "anonymous namespace" is supposed to ensure LOGNAME is local to this file.

`Having something in an anonymous namespace means it's local to this translation unit (.cpp file and all its includes) this means that if another symbol with the same name is defined elsewhere there will not be a violation of the One Definition Rule (ODR).`

But yeah, it's not working right. I think the problem might be using it in a header file.